Lib/test/test_py_compile.py

```diff
@@ -98,6 +98,20 @@ def test_bad_coding(self):
         self.assertFalse(os.path.exists(
             importlib.util.cache_from_source(bad_coding)))
 
+    def test_source_date_epoch(self):
+        os.environ["SOURCE_DATE_EPOCH"] = "123456789"
+        py_compile.compile(self.source_path, self.pyc_path)
+        del os.environ["SOURCE_DATE_EPOCH"]
+        self.assertTrue(os.path.exists(self.pyc_path))
+        self.assertFalse(os.path.exists(self.cache_path))
+        statinfo = os.stat(self.source_path)
+        self.assertEqual(statinfo.st_mtime, 123456789)
+        f = open(self.pyc_path, "rb")
+        f.read(4)
+        timebytes = f.read(4) # read timestamp from pyc header
+        f.close()
+        self.assertEqual(timebytes, b'\x15\xcd\x5b\x07')
```

Comment:
Just to make this test a little more obvious, how about changing the bytes literal to `(123456789).to_bytes(4, 'little')` ?